### PR TITLE
[VyTT][CreatePost]: add create post component

### PR DESCRIPTION
**Pull request by tranVyTDT**

### CODE REVIEW

File: vue-app/src/views/Post/CreatePost.vue [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+                    <div class="row g-2 g-sm-4">
+                        <div class="col-6">
+                            <select class="form-select js-choice border-0 bg-light" aria-label=".form-select-sm" v-model="postData.totalExpectedTeachingTime">
+                                <option value="" disabled selected>Chọn thời gian</option>
+                                <option v-for="amount in amountMonth" :key="amount" :value="amount">{{ amount }}</option>
+                            </select>
+                        </div>
+                        <div class="col-6">
+                            <select class="form-select js-choice border-0 bg-light" aria-label=".form-select-sm" v-model="postData.totalExpectedTeachingUnit">
+                                <option value="" disabled selected>Chọn thời gian</option>
```
Comment (rom2kpro) [inline vue-app/src/views/Post/CreatePost.vue:271]:
```suggestion
                                <option value="" disabled selected>Đơn vị</option>
```

File: vue-app/src/service/common.service.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type { CourseInterface } from '@/interfaces/common.interface'
+
+class CommonService {
+    getCourses() {
+        const mockup_data: CourseInterface[] = [
```
Comment (rom2kpro) [inline vue-app/src/service/common.service.ts:5]:
```suggestion
        const mockup_data: TreeNode[] = [
```

File: vue-app/src/service/common.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                            {key: 3, label: 'test name 3', data: 'test title 1'},
+                            {key: 4, label: 'test name 4', data: 'test title 1'}
+                        ]
+                    },
+                    {key: 5, label: 'test name 5', data: 'test title 1', children: [
+                        {key: 6, label: 'test name 6', data: 'test title 1', children: [
+                            {key: 7, label: 'test name 7', data: 'test title 1'}
+                        ]},
+                        {key: 8, label: 'test name 8', data: 'test title 1'}
+                    ]},
```
Comment (rom2kpro) [inline vue-app/src/service/common.service.ts:18]: change from number to string like key: '1'

Comment (tranVyTDT) [reply vue-app/src/service/common.service.ts:18]: its mockup data will be replaced

File: vue-app/src/views/Post/CreatePost.vue [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+                </div>
+            </div>
+        </div>
+
+        <!-- Button -->
+        <div class="col-12 text-sm-end">
+            <button class="btn btn-primary mb-0">Submit</button>
+        </div>
+    </form>
+    <!-- Form END -->
```
[the remark this answers is not in the document]
Comment (tranVyTDT) [reply vue-app/src/views/Post/CreatePost.vue:417]: when i integrate create post api, i will finish those missing too

**Pull request merged**